app/src/main/java/com/example/tasky/agenda/data/sync/DeleteAgendaItemWorker.kt

```diff
@@ -0,0 +1,100 @@
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.tasky.agenda.domain.data.network.TaskRemoteDataSource
+import com.example.tasky.agenda.domain.util.AgendaKind
+import com.example.tasky.core.data.database.SyncOperation
+import com.example.tasky.core.data.database.task.dao.TaskPendingSyncDao
+import com.example.tasky.core.domain.util.Result.Error as ResultError
+import com.example.tasky.core.domain.util.Result.Success as ResultSuccess
+
+class DeleteAgendaItemWorker(
+    context: Context,
+    params: WorkerParameters,
+    private val taskRemoteDataSource: TaskRemoteDataSource,
+    private val taskPendingSyncDao: TaskPendingSyncDao,
+) : CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        if (runAttemptCount >= 5) {
+            return Result.failure()
+        }
+
+        val id = inputData.getString(ITEM_ID) ?: return Result.failure()
+        val kind = inputData.getString(ITEM_KIND)
+            ?.let { runCatching { AgendaKind.valueOf(it) }.getOrNull() }
+            ?: return Result.failure()
+
+        return when (kind) {
+            AgendaKind.TASK -> {
+                deleteTask(itemId = id)
+            }
+
+            AgendaKind.EVENT -> {
+                deleteEvent(itemId = id)
+            }
+
+            AgendaKind.REMINDER -> {
+                deleteReminder(itemId = id)
+            }
+        }
+    }
+
+    private suspend fun deleteTask(itemId: String): Result {
+        val pendingCreateTask = taskPendingSyncDao.getTaskPendingSyncEntityByIdAndOperation(
+            taskId = itemId, operation = SyncOperation.CREATE
+        )
+        val pendingUpdateTask = taskPendingSyncDao.getTaskPendingSyncEntityByIdAndOperation(
+            taskId = itemId, operation = SyncOperation.CREATE
+        )
+
+        if (pendingCreateTask != null && pendingUpdateTask != null) {
+            taskPendingSyncDao.deleteTaskPendingSyncEntity(
+                taskId = itemId,
+                operations = listOf(SyncOperation.CREATE, SyncOperation.UPDATE)
+            )
+            taskPendingSyncDao.deleteDeletedTaskSyncEntity(taskId = itemId)
+
+            return Result.success()
+        } else if (pendingCreateTask != null) {
+            taskPendingSyncDao.deleteTaskPendingSyncEntity(
+                taskId = itemId,
+                operations = listOf(SyncOperation.CREATE)
+            )
+            taskPendingSyncDao.deleteDeletedTaskSyncEntity(taskId = itemId)
+
+            return Result.success()
+        } else if (pendingUpdateTask != null) {
+            taskPendingSyncDao.deleteTaskPendingSyncEntity(
+                taskId = itemId,
+                operations = listOf(SyncOperation.CREATE)
+            )
+        }
+
+        return when (val result = taskRemoteDataSource.deleteTask(id = itemId)) {
+            is ResultError -> {
+                result.error.toWorkerResult()
```

Comment:
We might want to explicitly handle the 404 case here-  if the server reports “not found,” that means the item never existed remotely, so we should treat it as a success and just clear it locally.